include/mbedtls/ccm.h

```diff
@@ -316,6 +316,7 @@ int mbedtls_ccm_star_auth_decrypt( mbedtls_ccm_context *ctx, size_t length,
  *
  * \return          \c 0 on success.
  * \return          \#MBEDTLS_ERR_CCM_BAD_INPUT on failure:
+ *                  the context has not been initialized,
```

Comment:
It's impossible to detect that the context has not been initialized: it would require distinguishing whether the current memory content is due to calling `init` followed by other API functions vs uninitialized memory that just happens to contain exactly the same bit values. What we can detect (but might not detect all the time, since this is a low-level use-at-your-own-risk API) is whether the context is in the wrong state, e.g. attempting to call update_ad() after update(), or attempting to call either without starts(). In starts(), an error condition could be “the context has already been started”.

Reply:
My bad, in "Improve expected context state" commit, I shouldn't have changed the error code descriptions I guess. I am going to remove them all.